server/repository/src/migrations/v2_04_00/add_bundled_item_table.rs

```diff
@@ -0,0 +1,26 @@
+use crate::migrations::*;
+
+pub(crate) struct Migrate;
+
+impl MigrationFragment for Migrate {
+    fn identifier(&self) -> &'static str {
+        "add_bundled_item_table"
+    }
+
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        sql!(
+            connection,
+            r#"
+                CREATE TABLE bundled_item (
+                    id TEXT NOT NULL PRIMARY KEY,
+                    principal_item_variant_id TEXT NOT NULL REFERENCES item_variant(id),
```

Comment:
Should we add item_id NOT null and make this nullable?

Reply:
I've messaged Mark to double check from a product side, but I think my preferred design, if/when we end up with variants & bundling more integrated with the rest of OMS, would be to create a `default` variant for all items. That way we can handle items with and without many variants set up, without a bunch of forked logic and nullable fields to check... what do you reckon?

Reply:
> I'm a little bit concerned about calling this bundled items when it's really just bundled item variants.
Should we change it? Or if we think we'll support bundling items to items or variants to items in the future should we update the db design now so we have optional fields? It's a pain to change these afterwards...

Mark seemed happy with the default variant solution, if you are? And @andreievg?

Reply:
I think we can get away with not deciding for now? I'm thinking the default variant solution is probably a good one, but we don't need it yet right?  Maybe we should create an issue for the default Item variants?